tests/linalg_test.py

```diff
@@ -1620,6 +1620,51 @@ def testToeplitzConstructionWithKnownCases(self):
       [3, 2, 1, 4]], dtype=np.float32))
 
 
+  base_shapes_small = [1, 2, 3]
+  base_shapes_big = [10, 100]
+  null_space_shapes = list(itertools.chain.from_iterable([[
+    (2, base_shape),
+    (base_shape, 2),
+    (1 + base_shape // 2, base_shape),
+  ] for base_shape in base_shapes_small + base_shapes_big]))
```

Comment:
We shouldn't be defining unnecessary class-level variables. You could put this logic inline, but that's probably not necessary. Couldn't we just iterate over a selection of shapes rather than constructing shapes with a particular structure? `null_space` should work for arbitrary 2D shapes, right? What if we do something like
```
@jtu.sample_product(
    shape=[[2, 1], [1, 2], [2, 5], [5, 2], [4, 20], [20, 4]],
    dtype=float_types + complex_types)
```
Do you think that would be sufficient test coverage?

Reply:
yes I think it would be, I just didn't want to make any arbitrary decisions and I thought that the less arbitrary was to just reuse the shapes scipy was testing, but I agree that going for something simpler is also a good point

Reply:
done